src/collection/Queue.php

```diff
@@ -40,7 +38,7 @@ public function __construct($collection = []) {
 	 */
 	public function enqueue(...$elements): self {
 		foreach ($elements as $element) {
-			array_unshift($this->array, $element);
+			array_push($this->array, $element);
```

Comment:
I think this methods body will just be `array_unshift($this->array, ...$elements)` (without the loop)

is it?

Reply:
Fixed. Now tests are ok.